python/cudf/cudf/core/series.py

```diff
@@ -831,62 +831,26 @@ def reindex(self, index=None, copy=True):
         series.name = self.name
         return series
 
-    def reset_index(self, drop=False, inplace=False):
-        """
-        Reset index to RangeIndex
-
-        Parameters
-        ----------
-        drop : bool, default False
-            Just reset the index, without inserting it as a column in
-            the new DataFrame.
-        inplace : bool, default False
-            Modify the Series in place (do not create a new object).
-
-        Returns
-        -------
-        Series or DataFrame or None
-            When `drop` is False (the default), a DataFrame is returned.
-            The newly created columns will come first in the DataFrame,
-            followed by the original Series values.
-            When `drop` is True, a `Series` is returned.
-            In either case, if ``inplace=True``, no value is returned.
-
-        Examples
-        --------
-        >>> import cudf
-        >>> series = cudf.Series(['a', 'b', 'c', 'd'], index=[10, 11, 12, 13])
-        >>> series
-        10    a
-        11    b
-        12    c
-        13    d
-        dtype: object
-        >>> series.reset_index()
-           index  0
-        0     10  a
-        1     11  b
-        2     12  c
-        3     13  d
-        >>> series.reset_index(drop=True)
-        0    a
-        1    b
-        2    c
-        3    d
-        dtype: object
-        """
+    @docutils.doc_series_reset_index()
+    def reset_index(self, level=None, drop=False, name=None, inplace=False):
+        """{docstring}"""
+        data, index = self._reset_index(level=level, drop=drop)
         if not drop:
             if inplace is True:
                 raise TypeError(
                     "Cannot reset_index inplace on a Series "
                     "to create a DataFrame"
                 )
-            return self.to_frame().reset_index(drop=drop)
-        else:
-            if inplace is True:
-                self._index = RangeIndex(len(self))
-            else:
-                return self._from_data(self._data, index=RangeIndex(len(self)))
+            if name is None:
+                name = 0 if self.name is None else self.name
+            data[name] = data.pop(self.name)
+            return cudf.core.dataframe.DataFrame._from_data(data, index)
+        # For ``name`` behavior, see:
+        # https://github.com/pandas-dev/pandas/issues/44575
```

Comment:
Can we write a test that will fail when the pandas fix for this (already merged to their main branch) gets released into a version that we test with?

Reply:
Our current behavior matches pandas and is covered in `test_series.py::test_reset_index` with `drop=True`, `inplace=True`.  When we update pandas version this test should naturally fail.